### PR TITLE
CLI script to gather environment info

### PR DESCRIPTION
**Pull request by BramVanroy**
I noticed that all too often people leave the "Environment" section in their issue empty. However, things such as the version number of PT/TF and `transformers` itself are very useful to know when trying to debug things.

This PR adds a small script to the existing CLI workflow. Running `python transformers-cli info` will output something like this:

```
Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 2.4.0
- Platform: Windows-10-10.0.18362-SP0
- Python version: 3.6.8
- PyTorch version (GPU?): 1.4.0 (True)
- Tensorflow version (GPU?): 2.1.0 (True)
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

Note that GPUs being available through the DL framework (`GPU?`) are retrieved automatically, but that users still have to specify whether or not they are actually using the GPU.

In addition, the relevant issue templates have been updated to direct users to the script.

### CODE REVIEW

**Comment by LysandreJik** [comment]
This is really cool!

**Comment by BramVanroy** [comment]
> This is really cool!

Props to spaCy, since I basically stole [the idea](https://github.com/explosion/spaCy/blob/master/spacy/cli/info.py) from them.

**Comment by julien-c** [comment]
LGTM but I've also pinged @mfuntowicz as he will have good insight

**Comment by BramVanroy** [comment]
Is there a way to see which test are run in `check_code_quality`? I'm curious as to why it fails.

**Comment by LysandreJik** [comment]
It fails because the file `/home/circleci/transformers/src/transformers/commands/info.py` would be reformatted by black.

You can run `make style` at the root to set everything to black style.

**Comment by BramVanroy** [comment]
> It fails because the file `/home/circleci/transformers/src/transformers/commands/info.py` would be reformatted by black.
> 
> You can run `make style` at the root to set everything to black style.

Thanks. Is that different from running `black .`? I did that, and it formats all files (not only info.py).

**Comment by LysandreJik** [comment]
The black command we used uses a specific line-length which is different to the default (we use a line length of 119, we like it better). We also set it to be based on Python 3.5.

I believe the test now fails because of isort; that's weird, it should have been triggered by the `make style` as well and should have fixed the imports on its own.

**Comment by BramVanroy** [comment]
> The black command we used uses a specific line-length which is different to the default (we use a line length of 119, we like it better). We also set it to be based on Python 3.5.
> 
> I believe the test now fails because of isort; that's weird, it should have been triggered by the `make style` as well and should have fixed the imports on its own.

I am now manually running black/isort (without the 'check' flag) and pushing those commits in hopes that the tests will then pass. But, correct me if I'm wrong, isn't circleCI supposed to apply these runs (black, isort etc) before running the tests?

**Comment by julien-c** [comment]
Did you install isort with the exact version that's pinned in `CONTRIBUTING.md`?

If you do, both `make style` and `make quality` should reliably pass.

File: src/transformers/commands/info.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+            tf_version = tf.__version__
+            try:
+                # deprecated in v2.1
+                tf_cuda_available = tf.test.is_gpu_available()
+            except AttributeError:
+                # returns list of devices, convert to bool
+                tf_cuda_available = bool(tf.config.list_physical_devices("GPU"))
+        except ImportError:
+            pass
```
Comment (mfuntowicz) [inline src/transformers/commands/info.py]: _Suggestion_ -> rely on transformers methods to check platform availability:

```python
from transformers import is_torch_available, is_tf_available

if is_torch_available():
...

if is_tf_available():
...
```

**Comment by mfuntowicz** [comment]
It's a very cool feature to provide through the CLI.

I may suggest to rename the command from `info` to `env` as we may want to keep `info` for exposing information about models through cards / config.

What do you think ? @BramVanroy @julien-c @LysandreJik

File: src/transformers/commands/info.py (hunk 1 above)
Comment (BramVanroy) [reply src/transformers/commands/info.py]: Thanks, that make sense. Changed in https://github.com/huggingface/transformers/pull/2699/commits/f0d5aded94b2e6e80ca31ace8fc6ce82aba75ab9

**Comment by BramVanroy** [comment]
> Did you install isort with the exact version that's pinned in `CONTRIBUTING.md`?
> 
> If you do, both `make style` and `make quality` should reliably pass.

Ah, I missed the note on isort. Fixed it now.


> It's a very cool feature to provide through the CLI.
> 
> I may suggest to rename the command from `info` to `env` as we may want to keep `info` for exposing information about models through cards / config.
> 
> What do you think ? @BramVanroy @julien-c @LysandreJik

Good suggestion and future-proof! I renamed the CLI method to env, so the full command is 

```
python transformers-cli env
```

Issue templates have been update and this command has been added to CONTRIBUTING.md as well.

**Comment by thomwolf** [comment]
LGTM, thanks a lot @BramVanroy

**Pull request merged**